_posts/2016-10-05-writing-fast-rails.md

```diff
@@ -0,0 +1,133 @@
+---
+layout: post
+title:  "Tips for Writing Fast Rails: Part 1"
+date: 2016-10-05 08:37:00
+categories: ["performance", "rails"]
+author: "etagwerker"
+---
+
+[Rails](http://rubyonrails.org/) is a powerful framework. You can write a lot  
+of features in a short period of time. In the process you can easily
+write code that **performs poorly**.
+
+At [Ombu Labs](http://www.ombulabs.com) we like to [maintain Ruby on Rails
+applications](http://www.ombulabs.com/blog/tags/maintenance). In the process
+of maintaining them, adding features and fixing bugs, we like to improve
+the code and its performance (because we are [good *boy scouts*](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule)!)
+
+Here are some tips based on our experience.
+
+## Prefer `where` instead of `select`
+
+When you are performing a lot of calculations, you should load as little as
+possible into memory. **Always** prefer a SQL query vs. an object's method call.
+
+<!--more-->
+
+With [ActiveRecord](http://guides.rubyonrails.org/active_record_basics.html), it's
+easy to forget which methods load
+[ActiveRecord::Base](https://github.com/rails/rails/blob/master/activerecord/lib/active_record/base.rb)
+objects into memory and which perform a simple query instead.
+
+**The bigger the table, the slower the object load**. If you have a table with 80
+columns (**sigh!**), loading each record will take a lot longer than a table
+with 3 columns. So, you must  avoid object loads as much as possible. Only load
+objects into memory when you **really need them**.
+
+For example:
+
+    shop_ids.map do |shop_id|
+      products.select { |p| p.shop_id == shop_id }.first
+    end
```

Comment:
@etagwerker just wondering, is this properly Ruby syntax-highlighted or just shown as a text block? Instead of using spaces, what happens if you surround the code with <code>```ruby</code>? I think we have support for it.